crates/biome_deserialize/src/diagnostics.rs

```diff
@@ -5,8 +5,47 @@ use biome_diagnostics::{
     Advices, Diagnostic, DiagnosticTags, LogCategory, MessageAndDescription, Severity, Visit,
 };
 use biome_rowan::{SyntaxError, TextRange};
+use bitflags::bitflags;
 use serde::{Deserialize, Serialize};
 
+bitflags! {
+    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+    pub struct ExpectedType: u8 {
+        const NULL = 1 << 0;
+        const BOOL = 1 << 1;
+        const NUMBER = 1 << 2;
+        const STR = 1 << 3;
+        const ARRAY = 1 << 4;
+        const MAP = 1 << 5;
+    }
+}
+
+impl Display for ExpectedType {
+    fn fmt(&self, fmt: &mut biome_console::fmt::Formatter) -> std::io::Result<()> {
+        if self.is_empty() {
+            return write!(fmt, "no value");
+        }
+        let mut is_not_first = false;
+        for expected_type in self.iter() {
```

Comment:
The final string?

Reply:
Yes. If we expect multiple types, the error message should have something like this `Expected string, or number, or boolean but received ... `

We should make sure that the message is correctly formatted.

Reply:
I think I should implement `std::fmt::Display` instead of `biome_console::Display`. In the same way I should certainly use `impl std::fmt::Display` in all `DeserialziationDiagnostic` function? Not sure. A type that implements `std::fmt::Display` should implement `biome_console::Display`?

Reply:
I find a workaround by using `format_args!`.